src/V3AstNodeOther.h

```diff
@@ -1798,19 +1798,48 @@ class AstUdpTable final : public AstNode {
 public:
     AstUdpTable(FileLine* fl, AstUdpTableLine* linesp)
         : ASTGEN_SUPER_UdpTable(fl) {
-        addLinesp(linesp);
+        this->addLinesp(linesp);
+        if (!v3Global.hasTable()) v3Global.setHasTable();
     }
     ASTGEN_MEMBERS_AstUdpTable;
 };
 class AstUdpTableLine final : public AstNode {
-    string m_text;
+    // @astgen op1 := iFieldp : List[AstUdpTableLineVal] // Input fields
+    // @astgen op2 := oFieldp : List[AstUdpTableLineVal] // Output fields
+private:
+    const bool m_udpIsCombo;  // Combinational or sequential UDP
 
 public:
-    AstUdpTableLine(FileLine* fl, const string& text)
+    class UdpCombo {};
+    AstUdpTableLine(UdpCombo, FileLine* fl, AstUdpTableLineVal* iFieldp,
+                    AstUdpTableLineVal* oFieldp)
         : ASTGEN_SUPER_UdpTableLine(fl)
-        , m_text{text} {}
+        , m_udpIsCombo{true} {
+        this->addIFieldp(iFieldp);
+        this->addOFieldp(oFieldp);
```

Comment:
```suggestion
        addIFieldsp(iFieldp);
        addOFieldsp(oFieldp);
```

Don't need `this` though I know the source is inconsistent.